File: ios/Classes/BifrostCommonChannel.swift
//
//  BifrostCommonChannel.swift
//  bifrost
//
//  Created by Matheus Lutero on 23/02/21.
//

import Flutter
import Foundation

internal class BifrostCommonChannel {
    
    private var channel: FlutterMethodChannel!
    
    init(_ messenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(name: "bifrost/common", binaryMessenger: messenger)
    }
    
    func setMethodCallHandler(_ handler: @escaping FlutterMethodCallHandler) {
        channel.setMethodCallHandler(handler)
    }
}

File: android/src/main/kotlin/br/com/dextra/bifrost/BifrostCommonChannel.kt
package br.com.dextra.bifrost

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

internal class BifrostCommonChannel(messenger: BinaryMessenger, handler: MethodChannel.MethodCallHandler) {

  private val channel by lazy { MethodChannel(messenger, "bifrost/common") }

  init {
    channel.setMethodCallHandler(handler)
  }
}
